### PR TITLE
Logging improvements in X509AuthenticationUtil and X509ZNodeGroupAclP…

### PR DESCRIPTION
**Pull request by abhilash1in**
-Class names and method names can be included in logs automatically via logging configuration. Hence including hardcoded classnames and method names in logging statements is not necessary.

### CODE REVIEW

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java [hunk 1]
```diff
@@ -103,14 +103,14 @@ public static X509TrustManager createTrustManager(ZKConfig config) {
...
-        LOG.warn("X509AuthenticationUtil::Truststore not specified for client connection");
+        LOG.warn("Truststore location not specified for SSL");
       } else {
         try {
           tm = X509Util
               .createTrustManager(trustStoreLocation, trustStorePassword, trustStoreTypeProp,
                   crlEnabled, ocspEnabled, hostnameVerificationEnabled, false);
         } catch (X509Exception.TrustManagerException e) {
-          LOG.error("X509AuthenticationUtil::Failed to create trust manager", e);
+          LOG.error("Failed to create trust manager", e);
```
Comment (rahulrane50) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:113]: neat: "Failed to create trust manager :"

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java [hunk 2]
```diff
@@ -137,8 +137,7 @@ public static String getClientId(X509Certificate clientCert) {
       try {
         return X509AuthenticationUtil.matchAndExtractSAN(clientCert);
       } catch (Exception ce) {
-        LOG.warn("X509AuthenticationUtil::getClientId(): failed to match and extract a"
-            + " client ID from SAN! Using Subject DN instead...", ce);
+        LOG.warn("Failed to match and extract a client ID from SAN. Using Subject DN instead.", ce);
```
Comment (rahulrane50) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:140]: Same as above.

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java [hunk 3]
```diff
@@ -174,39 +173,32 @@ private static String matchAndExtractSAN(X509Certificate clientCert)
...
+      // See GeneralName object defined in RFC 5280 (The ASN.1 definition of the SubjectAltName extension)
+      String errStr = "Client cert ID type 'SAN' was provided but matchType or matchRegex given is invalid! "
+          + "matchType: " + matchType + " matchRegex: " + matchRegex;
       LOG.error(errStr);
       throw new IllegalArgumentException(errStr);
     }
     // filter by match type and match regex
-    LOG.info("X509AuthenticationUtil::matchAndExtractSAN(): number of SAN entries found in"
-        + " clientCert: " + clientCert.getSubjectAlternativeNames().size());
+    LOG.debug("Number of SAN entries found in client cert: " + clientCert.getSubjectAlternativeNames().size());
```
Comment (rahulrane50) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:188]: neat : space before ":"

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java [hunk 4]
```diff
@@ -174,39 +173,32 @@ private static String matchAndExtractSAN(X509Certificate clientCert)
...
     Pattern matchPattern = Pattern.compile(matchRegex);
     Collection<List<?>> matched = clientCert.getSubjectAlternativeNames().stream().filter(
         list -> list.get(0).equals(matchType) && matchPattern.matcher((CharSequence) list.get(1))
             .find()).collect(Collectors.toList());
 
-    LOG.info(
-        "X509AuthenticationUtil::matchAndExtractSAN(): number of SAN entries matched: " + matched
-            .size() + ". Printing all matches...");
+    LOG.debug("Number of SAN entries matched: " + matched.size() + ". Printing all matches...");
```
Comment (rahulrane50) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:194]: neat : space before ":". Also, "..." can be removed at the end of string

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java [hunk 5]
```diff
@@ -253,16 +246,16 @@ private void assignAuthInfo(ServerCnxn cnxn, String clientId, Set<String> domain
       // Remove all previously assigned ZNodeGroupAcls that are no longer valid.
       if (isZnodeGroupAclScheme(id.getScheme()) && !newAuthIds.contains(id)) {
         cnxn.removeAuthInfo(id);
-        LOG.info(logStrPrefix + "Authenticated Id 'scheme: {}, id: {}' has been removed from session 0x{}.",
-            id.getScheme(), id.getId(), Long.toHexString(cnxn.getSessionId()));
+        LOG.info("Authenticated Id 'scheme: {}, id: {}' has been removed from session 0x{} of host {}.",
+            id.getScheme(), id.getId(), Long.toHexString(cnxn.getSessionId()), cnxn.getHostAddress());
```
Comment (rahulrane50) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java:250]: Any reason why host address is removed?

**Review by rahulrane50 — COMMENTED** [review]
Thanks for working on it! Overall looks good to me just couple of formatting changes. like having space before ":" at all places

**Comment by abhilash1in** [comment]
> Thanks for working on it! Overall looks good to me just couple of formatting changes. like having space before ":" at all places

Adding a space before ":" is not a widely followed standard. Spaces are added **after** a ":". 

Quoting [Wikipedia](https://en.wikipedia.org/wiki/Colon_(punctuation)#Spacing):
> In modern [English-language](https://en.wikipedia.org/wiki/English-language) printing, no space is placed before a colon and a single space is placed after it.

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java (hunk 5 above)
Comment (abhilash1in) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java:250]: Host address was added, not removed.

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java (hunk 1 above)
Comment (abhilash1in) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:113]: The standard/convention is to **not** add ":" at the end of such logging statements.

Examples: 
- https://github.com/linkedin/zookeeper/blob/ade1f2686a9740f206c1bc7e33a791a6b6416400/zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java#L523
- https://github.com/linkedin/zookeeper/blob/li-dev/base-3.6.3-znode-group-security/zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java#L850
- https://github.com/linkedin/zookeeper/blob/li-dev/base-3.6.3-znode-group-security/zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java#L874

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java (hunk 2 above)
Comment (abhilash1in) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:140]: See https://github.com/linkedin/zookeeper/pull/83#discussion_r911476582

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java (hunk 3 above)
Comment (abhilash1in) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:188]: See https://github.com/linkedin/zookeeper/pull/83#issuecomment-1171717208

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java (hunk 4 above)
Comment (abhilash1in) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java:194]: See https://github.com/linkedin/zookeeper/pull/83#issuecomment-1171717208

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java (hunk 5 above)
Comment (rahulrane50) [reply zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java:250]: Oh my bad here

**Comment by rahulrane50** [comment]
Sure i would prefer that style but i agree that it's not industry standard.

**Review by rahulrane50 — APPROVED** [review]

**Pull request merged**